### ISSUE TITLE
Big performance issue with new chunks generation

### ISSUE THREAD

**Issue by uhloin** [CorgiTaco/BYG#1033]
### BYG Version

2.0.0.13

### Terrablender Version

2.0.1.130

### Mod Loader

Forge

### Mod Loader Version

43.1.57

### Bug Severity

Major

### What happened?

Conditions to reproduce
1. World should be big enough (1,5 Gb)
2. Need fast levitation like with elytras or creative fly or broom of evilcraft/hexerei.
3. Need move to border of generated world to force generation of new chunks.
4. To reproduce it is enought BYG, terrablender and rubidium.
The problem is freeze of generation of new chunks for minute or more without loading of CPU. Exploration of the world becomes a pain. Looks like 1 chunk on the border are generated but not shown I still can move there and stopped by invisible terrain. In 2 chunks from border movement is stopped too.

Of course fast levitation always cause performance troubles but... if I remove BYG, these freezes are disappeared and generation becomes as fast as lightning. 

I am sure there is no such a problem some time ago. But when I checked older versions of BYG and TerraBlender I could not found when it is started. All versions for 1.19.2 are slow. May be a problem is related to Forge changes. May be Rubidium (I use latest version). In any case BOP (biomes of plenty) has not such a problem. I use both BYG and BOP and to remove generation lags I need to remove only BYG.

### latest.log

-

### Did you isolate BYG?

I tried BYG alone

**Comment by CorgiTaco** [comment]
MC is slow in 1.18+ with world gen, pre-generate chunks, and set a world border to prevent new chunk generation. I'd also suggest starlight: https://www.curseforge.com/minecraft/mc-mods/starlight-forge

**Comment by uhloin** [comment]
Sorry but MC is not slow with world gen with Biomes of Plenty. And MC is not slow with world gen with Ars Nouveau (it also uses TerraBlender). It is slow only with BOP so I conclude something it is really slow but not MC :(

**Comment by uhloin** [comment]
After some experiments I got situation when performance was restored with latest versions of BYG/TB. So I am not sure what conditions leads to problems. Looks like some specific biomes are very slow and freezing but other make generation in appropriate time. For example "Embur Bog" may be problematic. If I save during freeze in Embur Bog, remove BYG and reload then many blocks disappeared and I have big improvement of performance.

I tried Starlight but it is buggy.
